map component, city label + some styling

jlurgo opened the pull request:
There's still some more work to do on getting the right responsive styling

![image](https://github.com/user-attachments/assets/23a4105f-b84c-40c2-9951-fd75f485ae35)
![image](https://github.com/user-attachments/assets/a6414770-2e86-4c5e-8dc8-a819012afb69)

closes #35

jlurgo committed:
map component, city label + some styling

chfritz commented:
Looks great! Can you add the mock simulator after all so I can try it out? Perhaps in a new directory `/test/mockBot`?

jlurgo committed:
added transMock

jlurgo commented:
Done! @chfritz

jlurgo committed:
renamed transMock to mockBot

jlurgo committed:
added dock and goto phillz service calls

jlurgo commented:
I added the service calls to make the robot dock and got to Phillz Coffee, implemented services in MockBot and refactored for housekeeping. I disabled getting current robot IP location for now, it starts docked and you can go back and forth to Phillz.

![image](https://github.com/user-attachments/assets/eb87c801-ef51-44fc-b1c7-8ee6f7e332d0)

chfritz commented on src/test/mockBot/mockBot.js:
Tests don't belong in `src`. They test the source. Please move `tests/` to top level.

jlurgo replied:
addressed

chfritz commented on src/test/mockBot/run.sh:
In node.js it is customary to use the `scripts` in package.json instead of adding bash scripts. Adding `node` to the path by borrowing it from Transitive is a bit weird because we need `npm` in the path already to run `npm i` beforehand. We should just add the `node` PATH to the container itself.

jlurgo replied:
Ok I added this script to packages.json and it works
"start": "bash -c 'source /opt/ros/noetic/setup.bash && node app.js'"

chfritz commented:
Did this break ("loading") because you removed the ipconfig logic? I'm running mockBot on homedesk.

![image](https://github.com/user-attachments/assets/3975b8a1-394f-454d-96d4-d39babc67181)

chfritz commented on src/client/components/map-component.tsx:
```diff
@@ -0,0 +1,97 @@
+log.setLevel('debug');
+
+
+
```
Doc comment please, explaining what this component does.

jlurgo replied:
done

chfritz commented on src/client/components/trigger-service-button.tsx:
```diff
@@ -0,0 +1,42 @@
+const log = getLogger('DockButton');
+log.setLevel('debug');
+
+
```
doc-comment please

jlurgo replied:
done

chfritz commented on src/client/components/city-label.tsx:
```diff
@@ -0,0 +1,33 @@
+log.setLevel('debug');
+
+const CITY_TOPIC = '/location/city';
+
```
doc-comment please

jlurgo replied:
done

jlurgo committed:
moved mockBot into test/

jlurgo committed:
Fixed topic for city label

jlurgo commented:
> Did this break ("loading") because you removed the ipconfig logic? I'm running mockBot on homedesk.
> 
> ![image](https://private-user-images.githubusercontent.com/4239098/384145307-3975b8a1-394f-454d-96d4-d39babc67181.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzEwNzk5NTgsIm5iZiI6MTczMTA3OTY1OCwicGF0aCI6Ii80MjM5MDk4LzM4NDE0NTMwNy0zOTc1YjhhMS0zOTRmLTQ1NGQtOTZkNC1kMzliYWJjNjcxODEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTEwOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMDhUMTUyNzM4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MjQwZmMyZGE1NmNkMDE5NGY4YTk2OTMzM2JmZjE3MTdlN2Y2YTRmMGQ2MTYyNzVkODAxMjkwM2U5ZjczMzE3NiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.vAlR02nhRLsrK9XxZn6KWWp9BZkipWWcohJ5QPeZSX0)

Fixed, the topic was not updated after the mock refactor

jlurgo committed:
added missing docstrings

jlurgo committed:
improved mockBot structure
